extensions/http/jetty/src/test/java/org/eclipse/dataspaceconnector/extension/jetty/JettyConfigurationTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.extension.jetty;
+
+import org.assertj.core.api.ThrowableAssert;
+import org.eclipse.dataspaceconnector.core.config.ConfigFactory;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class JettyConfigurationTest {
+
+
+    @Test
+    void createFromConfig_defaultPort() {
+
+        var res = JettyConfiguration.createFromConfig(null, null, ConfigFactory.fromMap(Map.of("web.http.port", "1234")));
+        assertThat(res.getPortMappings()).hasSize(1).allSatisfy(pm -> {
+            assertThat(pm.getName()).isEqualTo("default");
+            assertThat(pm.getPort()).isEqualTo(1234);
+        });
+    }
+
+    @Test
+    void createFromConfig_noPortFound() {
+        var res = JettyConfiguration.createFromConfig(null, null, ConfigFactory.fromMap(Map.of()));
+        assertThat(res.getPortMappings()).hasSize(1).allSatisfy(pm -> {
+            assertThat(pm.getName()).isEqualTo("default");
+            assertThat(pm.getPath()).isEqualTo("/api");
+            assertThat(pm.getPort()).isEqualTo(8181);
+        });
+
+    }
+
+    @Test
+    void createFromConfig_implicitDefaultAndAnotherPort() {
+        var res = JettyConfiguration.createFromConfig(null, null, ConfigFactory.fromMap(Map.of(
+                "web.http.port", "1234",
+                "web.http.another.port", "8888",
+                "web.http.another.path", "/foo/bar"
+        )));
+
+        assertThat(res.getPortMappings()).hasSize(2).anySatisfy(pm -> {
+            assertThat(pm.getName()).isEqualTo("default");
+            assertThat(pm.getPort()).isEqualTo(1234);
+            assertThat(pm.getPath()).isEqualTo("/api");
+        }).anySatisfy(pm -> {
+            assertThat(pm.getName()).isEqualTo("another");
+            assertThat(pm.getPort()).isEqualTo(8888);
+            assertThat(pm.getPath()).isEqualTo("/foo/bar");
+        });
+    }
+
+    @Test
+    void createFromConfig_implicitAndExplicitDefault_shouldThrowException() {
+        ThrowableAssert.ThrowingCallable doubleDefaultConfig = () -> JettyConfiguration.createFromConfig(null, null, ConfigFactory.fromMap(Map.of(
+                "web.http.port", "1234",
+                "web.http.default.port", "8888"
+        )));
+
+        assertThatThrownBy(doubleDefaultConfig).isInstanceOf(IllegalArgumentException.class).hasMessageStartingWith("A port mapping for port '8888' already exists");
```

Comment:
Is the message correct? the problem is that the "default" mapping could be defined, not that the port 8888 is already defined (because it isn't)